File: for-local-test/TypeScript/119.pascals-triangle-ii.ts
// cmd to run => npx ts-node 119.pascals-triangle-ii.ts
function getRow(rowIndex: number): number[] {
    const row: number[] = [];
    for (let i = 0; i <= rowIndex; i++) {
        row.push(1);
    }
    for (let i = 1; i <= rowIndex; i++) {
        for (let j = i - 1; j >= 1; j--) {
            row[j] = (row[j] ?? 0) + (row[j - 1] ?? 0);
        }
    }
    return row;
}

// Test
console.log(getRow(6));

File: for-local-test/python/119.pascals-triangle-ii.py
from typing import List


class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        # Initialize the row with 1's
        row = []
        for i in range(rowIndex + 1):
            row.append(1)
        for i in range(1, rowIndex + 1):
            for j in range(i - 1, 0, -1):
                row[j] = row[j] + row[j - 1]
        return row

if __name__ == "__main__":
    solution = Solution()
    print(solution.getRow(5))
